post_info.php

```diff
@@ -687,6 +717,100 @@ function get_choices_id(
     );
 }
 
+function add_new_survey_choice($survey_id, $level, $data, $user_id)
+{
+    // Use a loop to bind variables for choice 1 to 6
+    $choices_object = array();
+    $choices_array = array();
+    $department_name = null;
+    $course_code = null;
+    $section_id = null;
+    switch ($level) {
+        case "dept":
+            for ($i = 0; $i < 6; $i++) {
+                $choice_number = $i + 1;
+                $choice_name = "choice" . $choice_number;
+                $choices_object[$choice_name] = (int) $data[
+                    "dept_survey_choices"
+                ]["choices"][$i];
+            }
+            array_push($choices_array, $choices_object, null, null);
+            $department_name = $data["dept_survey_choices"]["department_name"];
```

Comment:
No need to reply to a comment if what you have to say is, "I will implement your comment". It causes needless emails to be sent...

Reply:
@siefkenj This is hard to simplify. It is different from the case we did last time: 1. It can just be exactly one of the three cases. 2. `ta_survey_choices` is different from `level=section`, we cannot directly use string concatenation in this case.

Reply:
@siefkenj An alternative solution is to only retrieve the `for` loop outside, reserving the `switch` statement at the same time.